migration/migrator/migrations/course/20220612122726_gradeable_specific_anon_ids.py

```diff
@@ -0,0 +1,62 @@
+"""Migration for a given Submitty course database."""
+import json
+from collections import OrderedDict
+
+def up(config, database, semester, course):
+    """
+    Run up migration.
+
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    :param database: Object for interacting with given database for environment
+    :type database: migrator.db.Database
+    :param semester: Semester of the course being migrated
+    :type semester: str
+    :param course: Code of course being migrated
+    :type course: str
+    """
+    database.execute(
+        """
+        CREATE TABLE IF NOT EXISTS gradeable_anon (
+            user_id character varying NOT NULL,
+            g_id character varying(255) NOT NULL,
+            anon_id character varying(255),
+            FOREIGN KEY(user_id)
+                REFERENCES users(user_id)
+                ON UPDATE CASCADE,
+            FOREIGN KEY(g_id)
+                REFERENCES gradeable(g_id)
+                ON UPDATE CASCADE
+        );
+        """
+    )
+    database_file = config.config_path / 'database.json'
+    with(database_file).open('r') as db_file:
+        db_info = json.load(db_file, object_pairs_hook=OrderedDict)
+        database.execute("GRANT SELECT, INSERT, UPDATE, DELETE ON gradeable_anon TO {}".format(db_info['database_course_user']))
+
```

Comment:
Ideally you shouldn't have to do this but I know it won't work otherwise. I just made an issue https://github.com/Submitty/Submitty/issues/8209 to track this. This is definitely correct for now but if that issue gets resolved before this goes in then I'd remove it.

Reply:
Okay :thumbsup: